### ISSUE TITLE
Zero-length DAO source files generated for non-updatable tables

### ISSUE THREAD

**Issue by sgrimm-sg** [jOOQ/jOOQ#4626]
Using jOOQ 3.7.0 with Java 8 and PostgreSQL 9.4. My schema has a few views. During jOOQ's code generation, I see, e.g.,

```
[INFO] Generating DAO           : FormSubmissionVersionsDao.java
[INFO] Generating DAO           : FormSubmissionsDao.java
[INFO] Skipping DAO generation  : FormSubmissionsDao.java
[INFO] Generating DAO           : FormTypesDao.java
```

In this case FormSubmissions is a view and skipping the DAO generation would be fine. But it's not actually skipped; after the build there is a zero-length `FormSubmissionsDao.java` in the generated sources directory. Since a zero-length source file doesn't compile, Maven always sees that `FormSubmissionsDao.class` is missing and recompiles even when there has been no schema change. I am using `<schemaVersionProvider>` which appears to be working as of 3.7.0.

**Comment by lukaseder** [comment]
Thank you very much for your report. Interesting observation about the recompilation with Maven. We hadn't noticed this yet - will check.

Zero-length source files should be perfectly OK with the Java compiler, but we don't have any reason to generate those, so we'll fix that ASAP.

**Comment by sgrimm-sg** [comment]
My observation about Maven was based on running `mvn -X compile` and seeing "Stale source detected" messages for all of the view DAO classes (but no other source files), then looking in the classes directory and seeing that there were no .class files for those view DAOs even after a successful build.

Thanks for the quick response! Once this is fixed, and with the fix to `<schemaVersionProvider>` in 3.7, our builds are going to get a lot faster.

**Comment by lukaseder** [comment]
All tables without primary keys are affected (but not necessarily views, e.g. when `<syntheticPrimaryKeys/>` or `<overridePrimaryKeys/>` is applied.
